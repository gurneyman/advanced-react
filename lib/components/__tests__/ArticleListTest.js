import React from 'react';
import ArticleList from '../ArticleList';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('ArticleList', () => {

  const testProps = {
    articles: {
      a: { id: 'a', body: 'hi', date: 'datestr', title: 'title a' },
      b: { id: 'b', body: 'hi', date: 'datestr', title: 'title b' },
    },
  };

  it('renders correctly', () => {
    const wrapper = shallow(
      <ArticleList
        {...testProps}
      />
    );

    expect(wrapper.find('ArticleContainer').length).toBe(2);

    expect(wrapper).toMatchSnapshot();
  });

});