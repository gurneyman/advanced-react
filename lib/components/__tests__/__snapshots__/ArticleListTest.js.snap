// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleList
    articles={
      Object {
        "a": Object {
          "body": "hi",
          "date": "datestr",
          "id": "a",
          "title": "title a",
        },
        "b": Object {
          "body": "hi",
          "date": "datestr",
          "id": "b",
          "title": "title b",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ArticleContainer
          article={
            Object {
              "body": "hi",
              "date": "datestr",
              "id": "a",
              "title": "title a",
            }
          }
        />,
        <ArticleContainer
          article={
            Object {
              "body": "hi",
              "date": "datestr",
              "id": "b",
              "title": "title b",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "a",
        "nodeType": "class",
        "props": Object {
          "article": Object {
            "body": "hi",
            "date": "datestr",
            "id": "a",
            "title": "title a",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "b",
        "nodeType": "class",
        "props": Object {
          "article": Object {
            "body": "hi",
            "date": "datestr",
            "id": "b",
            "title": "title b",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ArticleContainer
            article={
              Object {
                "body": "hi",
                "date": "datestr",
                "id": "a",
                "title": "title a",
              }
            }
          />,
          <ArticleContainer
            article={
              Object {
                "body": "hi",
                "date": "datestr",
                "id": "b",
                "title": "title b",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "a",
          "nodeType": "class",
          "props": Object {
            "article": Object {
              "body": "hi",
              "date": "datestr",
              "id": "a",
              "title": "title a",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "b",
          "nodeType": "class",
          "props": Object {
            "article": Object {
              "body": "hi",
              "date": "datestr",
              "id": "b",
              "title": "title b",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
